### PR TITLE
[mono] Loader improvements: bundle optimization and logging

### PR DESCRIPTION
**Pull request by CoffeeFlux**
See individual commits and description.

### CODE REVIEW

File: src/mono/mono/metadata/assembly.c [hunk 1]
```diff
@@ -1663,27 +1684,36 @@ netcore_load_reference (MonoAssemblyName *aname, MonoAssemblyLoadContext *alc, M
...
+	 * 9. Return NULL.
 	 */
 
 	reference = mono_assembly_loaded_internal (alc, aname, FALSE);
 	if (reference)
 		goto leave;
 
+	if (bundles != NULL) {
+		reference = search_bundle_for_assembly (alc, aname);
+		if (reference) {
```
Comment (mdh1418) [inline src/mono/mono/metadata/assembly.c:1714]: Remove curly braces?

File: src/mono/mono/metadata/assembly.c [hunk 2]
```diff
@@ -1642,6 +1642,27 @@ load_reference_by_aname_individual_asmctx (MonoAssemblyName *aname, MonoAssembly
...
+{
+	if (bundles == NULL)
+		return NULL;
+
+	MonoImageOpenStatus status;
+	MonoImage *image;
+	MonoAssemblyLoadRequest req;
+	image = mono_assembly_open_from_bundle (alc, aname->name, &status, FALSE);
+	if (!image) {
+		char *name = g_strdup_printf ("%s.dll", aname->name);
```
Comment (mdh1418) [inline src/mono/mono/metadata/assembly.c:1656]: Does `mono_assembly_open_from_bundle` work if `aname->name` doesn't have a `.dll` suffix? If it only works with `.dll` suffix then would it make sense to default to appending `.dll` if it doesn't have it so we would only call `mono_assembly_open_from_bundle` once?

File: src/mono/mono/metadata/assembly.c (hunk 1 above)
Comment (mdh1418) [reply src/mono/mono/metadata/assembly.c:1714]: Oh never mind, I didn't see commit 2 before making this comment

File: src/mono/mono/metadata/assembly.c (hunk 2 above)
Comment (CoffeeFlux) [reply src/mono/mono/metadata/assembly.c:1656]: It totally depends on how the name is registered in the bundle.

Our behavior has historically been to just try with the name directly and then, if it fails, try postfixing, which seems sane to me. You have to try both cases unless we want to break historical behavior (we don't), and trying the original name first seems optimal.

Worth noting I haven't changed this logic at all in the PR, just moved where it occurs.

File: src/mono/mono/metadata/assembly.c (hunk 1 above)
Comment (CoffeeFlux) [reply src/mono/mono/metadata/assembly.c:1714]: I couldn't be bothered to fix that in the original commit when chunking this up since it's getting squashed anyway :p

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/metadata/appdomain.c [hunk 3]
```diff
@@ -2480,7 +2480,7 @@ mono_domain_assembly_preload (MonoAssemblyLoadContext *alc,
 
 		char *base_dir = get_app_context_base_directory (error);
 		search_path [0] = base_dir;
-		mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_ASSEMBLY, "Domain %s (%p) ApplicationBase is %s", domain->friendly_name, domain, base_dir);
+		mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_ASSEMBLY, "Domain (%p) ApplicationBase is %s", domain, base_dir);
```
Comment (vargaz) [inline src/mono/mono/metadata/appdomain.c:2484]: Why is this needed ?

Comment (CoffeeFlux) [reply src/mono/mono/metadata/appdomain.c:2484]: On netcore we only have one domain and it's named 'mono', so this doesn't actually offer any useful information and looks dumb in the logs. Note that this log is in netcore-specific code and I haven't touched the legacy equivalent. If you think there's a reason to keep it I'm happy to, but otherwise might as well clean it up.
